src/libraries/screensavermanager.js

```diff
@@ -1,134 +1,128 @@
-define(["events", "playbackManager", "pluginManager", "inputManager", "connectionManager", "userSettings"], function (events, playbackManager, pluginManager, inputManager, connectionManager, userSettings) {
-    "use strict";
-
-    playbackManager = playbackManager.default || playbackManager;
+import events from 'events';
+import playbackManager from 'playbackManager';
+import pluginManager from 'pluginManager';
+import inputManager from 'inputManager';
+import connectionManager from 'connectionManager';
+import * as userSettings from 'userSettings';
+
+function getMinIdleTime() {
+    // Returns the minimum amount of idle time required before the screen saver can be displayed
+    //time units used Millisecond
+    return 180000;
+}
+
+let lastFunctionalEvent = 0;
+
+function getFunctionalEventIdleTime() {
+    return new Date().getTime() - lastFunctionalEvent;
+}
+
+events.on(playbackManager, 'playbackstop', function (e, stopInfo) {
+    const state = stopInfo.state;
+    if (state.NowPlayingItem && state.NowPlayingItem.MediaType == 'Video') {
+        lastFunctionalEvent = new Date().getTime();
+    }
+});
 
-    function getMinIdleTime() {
-        // Returns the minimum amount of idle time required before the screen saver can be displayed
-        //time units used Millisecond
-        return 180000;
+function getScreensaverPlugin(isLoggedIn) {
+    let option;
+    try {
+        option = userSettings.get('screensaver', false);
+    } catch (err) {
+        option = isLoggedIn ? 'backdropscreensaver' : 'logoscreensaver';
```

Comment:
_Not related to this PR_
It doesn't throw on unauthorized `userSettings.get` and we don't get screensaver.